components/ApiKeyInput.tsx

```diff
@@ -1,27 +1,26 @@
-import { useState, useEffect } from 'react';
+import { useMemo } from 'react';
 
-export default function Component({defaultKey = null, keyName}) {
-    const [apiKey, setApiKey] = useState(() => {
-        const storedKey = window.localStorage.getItem(keyName);
-        return storedKey || defaultKey || '';
-    });
+interface Props {
+    keyName: string;
+    defaultKey?: string;
+    onChange?: (key: string) => void;
+}
 
-    useEffect(() => {
-        window.localStorage.setItem(keyName, apiKey);
-    }, [apiKey, keyName]);
-
-    const handleAPIKeyChange = (event) => {
-        setApiKey(event.target.value);
-    };
+export default function Component({ keyName, defaultKey = "", onChange = () => null }: Props) {
```

Comment:
I set `() => null` as a default value for `onChange` which means by default, for instance for the `database` key, nothing is done.